/var/spool/slurmd/job214348/slurm_script: line 14: conda: command not found
Loading tensorflow-gpu/py3/2.3.1
  Loading requirement: cuda/10.1.2 nccl/2.5.6-2-cuda cudnn/7.6.5.32-cuda-10.1
    intel-compilers/19.0.4 openmpi/4.0.2-cuda
+ srun python -u train_fred.py
2021-03-11 18:01:02.090249: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2021-03-11 18:02:42.727576: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcuda.so.1
2021-03-11 18:02:42.808858: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties: 
pciBusID: 0000:1a:00.0 name: Tesla V100-SXM2-32GB computeCapability: 7.0
coreClock: 1.53GHz coreCount: 80 deviceMemorySize: 31.75GiB deviceMemoryBandwidth: 836.37GiB/s
2021-03-11 18:02:42.808918: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2021-03-11 18:02:42.823218: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcublas.so.10
2021-03-11 18:02:42.831070: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcufft.so.10
2021-03-11 18:02:42.836799: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcurand.so.10
2021-03-11 18:02:42.846071: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusolver.so.10
2021-03-11 18:02:42.852475: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusparse.so.10
2021-03-11 18:02:42.897380: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.7
2021-03-11 18:02:42.901589: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1858] Adding visible gpu devices: 0
2021-03-11 18:02:42.902324: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  SSE3 SSE4.1 SSE4.2 AVX AVX2 AVX512F FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-03-11 18:02:42.911178: I tensorflow/core/platform/profile_utils/cpu_utils.cc:104] CPU Frequency: 2500000000 Hz
2021-03-11 18:02:42.911757: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x563ea56d13d0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2021-03-11 18:02:42.911782: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2021-03-11 18:02:42.913821: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties: 
pciBusID: 0000:1a:00.0 name: Tesla V100-SXM2-32GB computeCapability: 7.0
coreClock: 1.53GHz coreCount: 80 deviceMemorySize: 31.75GiB deviceMemoryBandwidth: 836.37GiB/s
2021-03-11 18:02:42.913860: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2021-03-11 18:02:42.913880: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcublas.so.10
2021-03-11 18:02:42.913895: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcufft.so.10
2021-03-11 18:02:42.913909: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcurand.so.10
2021-03-11 18:02:42.913922: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusolver.so.10
2021-03-11 18:02:42.913936: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusparse.so.10
2021-03-11 18:02:42.913950: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.7
2021-03-11 18:02:42.917975: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1858] Adding visible gpu devices: 0
2021-03-11 18:02:42.918031: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2021-03-11 18:02:43.913848: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1257] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-03-11 18:02:43.913884: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1263]      0 
2021-03-11 18:02:43.913898: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1276] 0:   N 
2021-03-11 18:02:43.919938: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1402] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 30132 MB memory) -> physical GPU (device: 0, name: Tesla V100-SXM2-32GB, pci bus id: 0000:1a:00.0, compute capability: 7.0)
2021-03-11 18:02:43.969118: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x563ec35223d0 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2021-03-11 18:02:43.969146: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla V100-SXM2-32GB, Compute Capability 7.0
2021-03-11 18:02:43.969545: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
  0%|          | 0/7265 [00:00<?, ?it/s]  0%|          | 0/7265 [00:01<?, ?it/s]
Traceback (most recent call last):
  File "train_fred.py", line 154, in <module>
    loss,label,prediction = compute_apply_gradients(model,loss_f,a,x_topic,x,x_mask,y,y_mask,optimizer)
  File "/gpfslocalsup/pub/anaconda-py3/2020.02/envs/tensorflow-gpu-2.3.1/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 780, in __call__
    result = self._call(*args, **kwds)
  File "/gpfslocalsup/pub/anaconda-py3/2020.02/envs/tensorflow-gpu-2.3.1/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 823, in _call
    self._initialize(args, kwds, add_initializers_to=initializers)
  File "/gpfslocalsup/pub/anaconda-py3/2020.02/envs/tensorflow-gpu-2.3.1/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 697, in _initialize
    *args, **kwds))
  File "/gpfslocalsup/pub/anaconda-py3/2020.02/envs/tensorflow-gpu-2.3.1/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 2855, in _get_concrete_function_internal_garbage_collected
    graph_function, _, _ = self._maybe_define_function(args, kwargs)
  File "/gpfslocalsup/pub/anaconda-py3/2020.02/envs/tensorflow-gpu-2.3.1/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 3213, in _maybe_define_function
    graph_function = self._create_graph_function(args, kwargs)
  File "/gpfslocalsup/pub/anaconda-py3/2020.02/envs/tensorflow-gpu-2.3.1/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 3075, in _create_graph_function
    capture_by_value=self._capture_by_value),
  File "/gpfslocalsup/pub/anaconda-py3/2020.02/envs/tensorflow-gpu-2.3.1/lib/python3.7/site-packages/tensorflow/python/framework/func_graph.py", line 986, in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
  File "/gpfslocalsup/pub/anaconda-py3/2020.02/envs/tensorflow-gpu-2.3.1/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 600, in wrapped_fn
    return weak_wrapped_fn().__wrapped__(*args, **kwds)
  File "/gpfslocalsup/pub/anaconda-py3/2020.02/envs/tensorflow-gpu-2.3.1/lib/python3.7/site-packages/tensorflow/python/framework/func_graph.py", line 973, in wrapper
    raise e.ag_error_metadata.to_exception(e)
TypeError: in user code:


    TypeError: tf__compute_apply_gradients() takes 6 positional arguments but 9 were given

srun: error: r8i0n6: task 0: Exited with exit code 1
srun: Terminating job step 214348.0
